plugins/modules/data_context.py

```diff
@@ -0,0 +1,308 @@
+# limitations under the License.
+
+from ansible_collections.cloudera.cluster.plugins.module_utils.cm_utils import (
+    ClouderaManagerMutableModule,
+)
+from cm_client import DataContextsResourceApi, ApiDataContextList
+
+from cm_client import (
+    ClustersResourceApi,
+    ApiDataContext,
+)
+from cm_client.rest import ApiException
+from ansible_collections.cloudera.cluster.plugins.module_utils.data_context_utils import (
+    parse_data_context_result,
+)
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: data_context
+short_description: Create, update, or delete a data context
+description:
+  - Configure details of a specific data context.
+  - Create a new data context.
+  - Update an existing data context.
+  - Delete a data context.
+  - The module supports C(check_mode).
+author:
+  - "Ronald Suplina (@rsuplina)"
+requirements:
+  - cm-client >= 54
+options:
+  name:
+    description:
+      - The name of the data context.
+    type: str
+    required: yes
+  cluster:
+    description:
+      - The name of the Cloudera Manager cluster.
+    type: str
+    required: no
+  services:
+    description:
+      - A list of services that the data context will include.
+    type: list
+    required: no
+  state:
+    description:
+      - If I(state=present), the data context will be created or updated.
+      - If I(state=absent), the data context will be deleted
+    type: str
+    required: no
+    default: present
+    choices:
+      - present
+      - absent
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: full
+"""
+
+EXAMPLES = r"""
+---
+- name: Create a Data Context
+  cloudera.cluster.data_context
+    host: example.cloudera.com
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+    name: "base_services"
+    cluster: "example_cluster"
+    services: ['hive','atlas','hdfs','ranger']
+    state: present
+
+- name: Delete a data context
+  cloudera.cluster.data_context
+    host: example.cloudera.com
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+    name: "base_services"
+    state: absent
+
+- name: Update an existing data context
+  cloudera.cluster.data_context
+    host: example.cloudera.com
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+    name: "base_services"
+    cluster: "example_cluster"
+    services: ['hive','atlas','hdfs']
+    state: present
+"""
+
+RETURN = r"""
+---
+data_context:
+  description:
+    - A dictionary containing details of data contexts within the cluster.
+  type: dict
+  elements: complex
+  returned: always
+  contains:
+    name:
+      description:
+        - The name of the data context.
+      type: str
+      returned: always
+    display_name:
+      description:
+        - The display name of the data context.
+      type: str
+      returned: always
+    nameservice:
```

Comment:
Is this actually `nameservice` or should it be "Python-ized" to `name_service`?

Reply:
![Screenshot 2024-07-29 at 16 38 41](https://github.com/user-attachments/assets/b3bc6685-ca0b-4df3-bd28-b07238cd4c82)
It is actually `nameservice`